### ISSUE TITLE
Native not working

### ISSUE THREAD

**Issue by ericmuyser** [rofrischmann/react-look#193]
Hey guys,

I've been trying to get react-look working with either React Native v0.17 or v0.19 with no success. Problems include:

1. Error stating there are multiple versions of react in v0.17. Due to devDependencies. Fixed by removing react manually.
2. Error that react doesn't exist, due to above. Fixed by changing references to `react-native`. Not sure how to get around this generically.
3. Error stating `ua.match` is undefined in v0.17 and v0.19. Due to not having a user agent. In bowser.js and other places. Fixed by hardcoding my own ua.
4. Error stating it can't find react-look in a folder up. Due to `import look, { StyleSheet } from 'react-look'` not resolving properly. Fixed by doing `var look = require('react-look'); var { Enhancer, StyleSheet } = look;` (and exposing Enhancer).
5. Finally, everything runs without error. But it doesn't work. The value of the `styles.something` are strings. I assume they are class names and it's attempting to inject CSS. That doesn't work on native I assume, and so it does nothing. Where is the fallback for native that gets inline styles working? I can see you had native working back in october, but I'm unclear as to how it was working so I can get this going. Ideas?

Seems like some of this could be due to it detecting the wrong environment, which I probably didn't help by specifying my own UA.

Cheers

**Comment by ericmuyser** [comment]
Looks like I missed this crucial step: https://github.com/rofrischmann/react-look/blob/develop/docs/GettingStarted.md#61-configuration--lookroot

Now getting `Invariant violation: Expected a component class, got div. Each component name should start with an uppercase letter`.

Re-installing.. trying again.. 

Coming across issues with it looking for react again. Re-tracing previous steps.

**Comment by ericmuyser** [comment]
Ya so all I had to do was replace `react` imports with `react-native` to get it running. Then following the instructions about LookRoot, it trips on this:
```
      return _react2.default.createElement(
        'div',
        otherProps,
        this.props.children,
        _react2.default.createElement(StyleComponent, { userAgent: config.userAgent })
      );
```

If I replace the 1st argument, `div`, with `_react2.default.View` and remove the 4th argument, the style element, it loads, no errors no warnings. But it only shows for half a second then goes white. Some flex issue perhaps?

**Comment by rofrischmann** [comment]
@ericmuyser Heyhey, 
I am really sorry to tell you that 1.0.0-beta right now does not support React Native. I am also sorry for the time you spent trying to get it work right now, but perhaps I got some good news:
You could try with 0.7.2 which had support, but rather wait until the end of this week. I am working to get Native support back as fast as possible, but I was not able to ship it with the exact code base as I am now using classNames as you mentioned above. 
That's why I also removed React Native from the documentation (Only the native preset remained).
The new Native module will be much simpler, smaller and even faster. 

I will update you as soon I got it working fully.

**Comment by ericmuyser** [comment]
So `style={{flex: 1}}` on the LookRoot fixed the white flash. Actually runs without errors!

Just doesn't apply any styles.. 

![image](https://cloud.githubusercontent.com/assets/156768/12983368/d9c1ace4-d09d-11e5-8f38-4a28c9103388.png)

**Comment by ericmuyser** [comment]
@rofrischmann Oh man :( I based my decision to put in this effort based on it saying it did in NPM: https://www.npmjs.com/package/react-look + some tickets where you removed the deps and suggested it should still work. It would be awesome if you could put that in the readme. Looking forward to 2.0. Can I help? Cheers

**Comment by rofrischmann** [comment]
Oh damn. I get it. npm does not show the 1.0.0-beta. Actually the 0.7.2 Version which is listed on npm should support React Native. I actually already put a warning into the README. 
You do not have to wait until 2.0, I will actually ship native support with 1.0.0-Beta2 already. It will be coming at the end of the week hopefully. Then I could need some help testing and improving the native experience :)
And again sorry about that :(

**Comment by ericmuyser** [comment]
Alright, no worries man. I understand. Thanks for the help! 

Looks wicked. I'll wait for beta2 :+1:

**Comment by rofrischmann** [comment]
I was using --tag beta, should perhaps switch to just publishing 1.0.0-beta1 as version. Then npm would also show the newest version^^

**Comment by ericmuyser** [comment]
As a side suggestion, I've noticed a lot of module developers are not putting in much effort to specify the versions they have things working with. It's become very difficult to get things to play nice with the different versions of NPM, React, and Node. I would love if module devs would specify the verified combinations. NPM3 in particular is causing a lot of ongoing stress: https://github.com/facebook/react-native/issues/4968
Order of importance seems to be 1) NPM version, 2) React version, 3) Package dep versions, 4) Node version from time to time.

**Comment by rofrischmann** [comment]
Oh thanks for the suggestion. I can actually agree to that... I already messed up with the versions and deps a lot. 
I will try to provide such specs from now on. Still learning all the stuff^^

**Comment by rofrischmann** [comment]
@ericmuyser React Native was fully included with #200. I'll soon update to beta2
